src/qibo/models/dbi/double_bracket.py

```diff
@@ -156,6 +203,97 @@ def hyperopt_step(
         )
         return best["step"]
 
+    def polynomial_step(
+        self,
+        n: int = 3,
+        n_max: int = 5,
+        d: np.array = None,
+        backup_scheduling: DoubleBracketScheduling = None,
+    ):
+        r"""
+        Optimizes iteration step by solving the n_th order polynomial expansion of the loss function.
+        e.g. $n=2$: $2\Trace(\sigma(\Gamma_1 + s\Gamma_2 + s^2/2\Gamma_3)\sigma(\Gamma_0 + s\Gamma_1 + s^2/2\Gamma_2))
+        Args:
+            n (int, optional): The order to which the loss function is expanded. Defaults to 3.
+            n_max (int, optional): The maximum order allowed for recurring calls of `polynomial_step`. Defaults to 5.
+            d (np.array, optional): The diagonal operator, default as $\delta(H)$.
+        """
+
+        if d is None:
+            d = self.diagonal_h_matrix
+
+        if backup_scheduling is None:
+            backup_scheduling = DoubleBracketScheduling.use_grid_search
+
+        def sigma(h: np.array):
+            return h - self.backend.cast(np.diag(np.diag(self.backend.to_numpy(h))))
+
```

Comment:
Yes, but here the `h` is a general Hamiltonian instead of the property `h` in class `DoubleBracketIteration`.  Is there a good way to reduce the redundancy such that we can have both the property and a function that works for other hamiltonians?

Reply:
maybe the property in line 115 should actually call 228?

Reply:
I suggest you define a `sigma` static method inside `DoubleBracketIteration` class and call it inside the property `off_diag_h`